packages/mobile/src/harmony-native/components/CoverPhoto/CoverPhoto.tsx

```diff
@@ -51,20 +56,35 @@ export const CoverPhoto = (props: CoverPhotoProps) => {
   const fullHeightStyle = css({ height: '100%' })
 
   const getSource = () => {
-    // Having .url means its a useable image source
-    if (coverPhoto && !isEmpty(coverPhoto)) {
-      return { source: coverPhoto }
+    let source: CoverPhotoImage = {
+      uri: undefined
     }
+    let usingProfilePicture = false
     if (profilePicture && !isEmpty(profilePicture)) {
-      return { source: profilePicture, usingProfilePicture: true }
+      source = profilePicture
+      usingProfilePicture = true
+    }
+    if (coverPhoto && !isEmpty(coverPhoto)) {
+      source = coverPhoto
+      usingProfilePicture = false
+    }
+
+    // Android upload format does not quite match the expected format, so we have to drill into 'file' to workaround for android
+    if ('file' in source && !('uri' in source)) {
+      return { source: source.file, usingProfilePicture }
+    } else {
+      return { source, usingProfilePicture }
```

Comment:
Android seems to have a slightly different upload format. I don't love this solution, but it does the job. Open to ideas of better options